build: Move `message.{h,cpp}` from `libbitcoin_util` to `libbitcoin_common`

hebasto opened the pull request:
This PR breaks the `libbitcoin_util` library's dependency on the `libbitcoin_consensus` library due to the `CPubKey::RecoverCompact` symbol, which is not in alignment with our library design goals.

The `libbitcoin_util` library, unlike `libbitcoin_common`, is a dependency of the kernel library, so we are trying to move unnecessary things out of it, to make the kernel library smaller.

For more details please refer to https://github.com/bitcoin/bitcoin/issues/28548.

hebasto committed:
build: Move `message.{h,cpp}` from libbitcoin_util to libbitcoin_common

This change breaks the `libbitcoin_util` library's dependency on the
`libbitcoin_consensus` library due to the `CPubKey::RecoverCompact`
symbol, which is not in alignment with our library design goals.

ryanofsky commented:
Code review ACK 31eaa9db17b253683a0a457a85ae5240d3587cc2. If you wanted to update PR description to describe rationale a little more, you could say libbitcoin_util library, unlike libbitcoin_common, is a dependency of the kernel library, so we are trying to move unnecessary things out of it, to make the kernel library smaller

hebasto commented:
Updated 31eaa9db17b253683a0a457a85ae5240d3587cc2 -> e39d82720641317d7f24d6564c1286641e81b1e2 ([pr28549.01](https://github.com/hebasto/bitcoin/commits/pr28549.01) -> [pr28549.02](https://github.com/hebasto/bitcoin/commits/pr28549.02), [diff](https://github.com/hebasto/bitcoin/compare/pr28549.01..pr28549.02)):

- fixed omitted header renaming

@ryanofsky 
> f you wanted to update PR description to describe rationale a little more, you could say libbitcoin_util library, unlike libbitcoin_common, is a dependency of the kernel library, so we are trying to move unnecessary things out of it, to make the kernel library smaller

Thank you! Done.

BrandonOdiwuor reviewed (commented):
Review ACK e39d82720641317d7f24d6564c1286641e81b1e2 moving `message.{h, cpp}` from `libbitcoin_util` to `libbitcoin_common` breaking its dependency on the `libbitcoin_consensus` library

fanquake commented:
cc @TheCharlatan

hebasto commented:
Closing in favour of #28690.
